backend/module/chaos/serverexperimentation/rtds/rtds_cache.go

```diff
@@ -120,33 +189,26 @@ func setSnapshot(snapshotCache gcpCache.SnapshotCache, rtdsLayerName string, ups
 		}
 	}
 
-	runtimes := []gcpTypes.Resource{
-		&gcpDiscovery.Runtime{
-			Name: rtdsLayerName,
-			Layer: &pstruct.Struct{
-				Fields: fieldMap,
-			},
-		},
+	runtimeLayer := &pstruct.Struct{
+		Fields: fieldMap,
 	}
 
-	computedVersion, err := computeChecksum(runtimes)
+	computedVersion, err := computeChecksum(runtimeLayer)
 	if err != nil {
 		logger.Errorw("Error computing version", "error", err)
 		return err
 	}
 
-	currentSnapshot, _ := snapshotCache.GetSnapshot(upstreamCluster)
-	if !reflect.DeepEqual(currentSnapshot, gcpCache.Snapshot{}) {
-		currentVersion := currentSnapshot.GetVersion(gcpResource.RuntimeType)
-		if currentVersion == computedVersion {
+	currentSnapshotVersion, err := snapshotCache.GetSnapshotVersion(upstreamCluster)
+	if err == nil {
```

Comment:
are we fine with proceeding for cases when error is equal to nil here?

Reply:
The underlying API from GCP for snapshot lookups is a bit odd, it uses error to denote whether the snapshot was found in the cache. This is how we end up with this off-looking code. 

I think ideally it would have an API similar to map lookups, e.g. `func GetSnapshot(key string) (Snapshot, bool)`, but right now the only "error" case is that the key doesn't exist in the cache, so it's keep going. This is basically doing the exact same thing as the previous deepEquals check but without having to use reflection.